### DISCUSSION TITLE
mknod: /dev/net/tun: operation not permitted

### DISCUSSION THREAD

**Discussion by tyhawk** [haugene/docker-transmission-openvpn#2393] [Ideas]
Perhaps one for the troubleshooting section of the documentation.
if you encounter this error:
```
Creating TUN device /dev/net/tun
mknod: /dev/net/tun: Operation not permitted
```
you need to use `--cap-add=NET_ADMIN,mknod` and probably `--device /dev/net/tun`.

This took me 2 days to figure out, running the container using Podman in:
- Debian (VM / LXC)
- Archlinux (LXC)
- MicroOS (VM)
I've got it up running now in a LXC container, Archlinux on Proxmox.

**Comment by haugene** [comment]
How is your docker run / docker compose setup @tyhawk ?

See my comment here about mounting the tun device: https://github.com/haugene/docker-transmission-openvpn/issues/2389#issuecomment-1312603471

**Comment by tyhawk** [comment]
Here's my command line:
```
/usr/bin/podman run -d --replace -dt --name transmission-vpn --label io.containers.autoupdate=image -v /srv/transmission:/data:z -v /mnt/download:/mnt/download --env-file /etc/podman/transmission.env --cap-add=NET_ADMIN,mknod --device /dev/net/tun --log-driver journald --dns 85.203.37.1 --dns 85.203.37.2 -p 9091:9091 docker.io/haugene/transmission-openvpn:latest
``` 
I've tried running it without the `device` command line option, but that didn't work either.  My `/etc/podman/transmission.env` is (the relevant parts):
```
CREATE_TUN_DEVICE=true
OPENVPN_PROVIDER=XXX
OPENVPN_USERNAME=xxx
OPENVPN_PASSWORD=xxx
OPENVPN_OPTS=--inactive 3600 --ping 10 --ping-exit 60
WEBPROXY_ENABLED=false
LOCAL_NETWORK=xxx.xxx.xxx.xxx/24
HEALTH_CHECK_HOST=google.com
WEBPROXY_ENABLED=true
WEBPROXY_PORT=8118
TRANSMISSION_WEB_UI=kettu
TRANSMISSION_ALT_SPEED_DOWN=4092
TRANSMISSION_ALT_SPEED_ENABLED=true
TRANSMISSION_ALT_SPEED_TIME_BEGIN=540
TRANSMISSION_ALT_SPEED_TIME_DAY=127
TRANSMISSION_ALT_SPEED_TIME_ENABLED=true
TRANSMISSION_ALT_SPEED_TIME_END=1380
TRANSMISSION_ALT_SPEED_UP=256
TRANSMISSION_DOWNLOAD_DIR=/mnt/download/complete
TRANSMISSION_DOWNLOAD_QUEUE_ENABLED=true
TRANSMISSION_DOWNLOAD_QUEUE_SIZE=20
TRANSMISSION_INCOMPLETE_DIR=/mnt/download/incomplete
TRANSMISSION_INCOMPLETE_DIR_ENABLED=true
TRANSMISSION_SPEED_LIMIT_DOWN=8192
TRANSMISSION_SPEED_LIMIT_DOWN_ENABLED=true
TRANSMISSION_SPEED_LIMIT_UP=1024
TRANSMISSION_SPEED_LIMIT_UP_ENABLED=true
TRANSMISSION_WATCH_DIR=/mnt/download/watch
TRANSMISSION_WATCH_DIR_ENABLED=true
TRANSMISSION_SCRAPE_PAUSED_TORRENTS_ENABLED=false
TRANSMISSION_RATIO_LIMIT=1.5
TRANSMISSION_RATIO_LIMIT_ENABLED=true
TRANSMISSION_PAUSED=true
```
I'm running podman in a (priviliged) LXC container (ArchLinux) on Proxmox.

The ` TRANSMISSION_PAUSED=true`  setting I added after getting the container to work, but I'm trying to get my torrents to be added paused (--paused). It doesn't seem to work though, so the setting is probably incorrect. I also added the Privoxy settings later as well, so I added -p 8118:8118 to the command line too.

**Comment by uprightbass360** [comment]
Apologies for the reply on an old discussion, but google brought me here directly and to anyone who is struggling, this may help. In addition to the configuration above, you may also need to pass your tun device into your lxc from proxmox. The conf file in /etc/lxc/xxx.conf will need:

lxc.cgroup2.devices.allow: c 10:200 rwm
lxc.mount.entry: /dev/net dev/net none bind,create=dir
